File: household-ledger-mobile/lib/domain/entities/category.dart
import 'transaction.dart';

class Category {
  final int id;
  final int? groupId;
  final int createdBy;
  final String name;
  final TransactionType type;
  final String? color;
  final bool isDefault;
  final int? budgetAmount;
  final DateTime createdAt;
  final DateTime updatedAt;

  Category({
    required this.id,
    this.groupId,
    required this.createdBy,
    required this.name,
    required this.type,
    this.color,
    required this.isDefault,
    this.budgetAmount,
    required this.createdAt,
    required this.updatedAt,
  });

  Category copyWith({
    int? id,
    int? groupId,
    int? createdBy,
    String? name,
    TransactionType? type,
    String? color,
    bool? isDefault,
    int? budgetAmount,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Category(
      id: id ?? this.id,
      groupId: groupId ?? this.groupId,
      createdBy: createdBy ?? this.createdBy,
      name: name ?? this.name,
      type: type ?? this.type,
      color: color ?? this.color,
      isDefault: isDefault ?? this.isDefault,
      budgetAmount: budgetAmount ?? this.budgetAmount,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}

File: apps/api/src/domain/models/category.rs
use chrono::NaiveDateTime;
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use super::transaction::TransactionType;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Category {
    pub id: i64,
    pub group_id: Option<i64>,
    pub created_by: i64,
    pub name: String,
    #[sqlx(rename = "type")]
    pub transaction_type: TransactionType,
    pub color: Option<String>,
    pub is_default: bool,
    pub budget_amount: Option<i64>,
    #[sqlx(default)]
    pub created_at: NaiveDateTime,
    #[sqlx(default)]
    pub updated_at: NaiveDateTime,
}

